spec/components/schemas/content/instagram/optin_response.ts

```diff
@@ -0,0 +1,33 @@
+import { SchemaObject } from 'openapi3-ts';
+import { ref as baseRef } from '../base';
+import { createComponentRef } from '../../../../../utils/ref';
+
+const optInResponse: SchemaObject = {
+  type: 'object',
+  allOf: [{
+    $ref: baseRef,
+  }, {
+    type: 'object',
+    properties: {
+      type: {
+        type: 'string',
+        example: 'optin_response',
+      },
+      notificationRecipient: {
+        type: 'string',
+        description: 'A token that you will be using to send recurring messages for a specific topic that user has opted-in. You will receive a unique token for each topic user opts-in. This attribute must be sent as `TO` attribute when sending a recurring notification message.',
+      },
+      optInDuration: {
```

Comment:
Duration não é um bom nome pois nessa campo não vai o tempo de vida e sim o momento de expiração. Que tal chamar de `optInExpiration`?

Quanto ao texto, o que acha dessa refatorada:
```
The timestamp when the opt-in will expire and no longer be usable, unless renewed.
<br>Just before this expiration, the user will have the option to renew this topic opt-in. When the user makes a choice, a new [webhook](#section/MESSAGE) with a new [OptIn Response](#section/OptIn-Response) will be sent.
```

Eu atualizei desse jeito pois vi que também vai webhook caso não seja renovado o opt-in. E também vem um indicado quando é feito um novo opt-in após um opt-out.

Então acho que seria legal a inclusão de mais um campo: `optInStatus` com os possíveis valores `OPTIN`, `OPTOUT`, e `REOPTIN`? E aí o texto poderia ser algo do tipo:
```
Indicates whether this is a new opt-in, an opt-out, or a reopt-in.
```

Reply:
Então, em pensei nesse campo novo ser justamente a combinação do `user_token_status` com o `notification_messages_status`:
- `OPTIN`: `REFRESHED` + `null`
- `OPTOUT`: `NOT_REFRESHED` | `STOP_NOTIFICATIONS`
- `REOPTIN`: `REFRESHED` + `RESUME_NOTIFICATIONS`